src/ui/test_ui.cpp

```diff
@@ -1,20 +1,52 @@
 #include "test_ui.h"
 
 #include "util/colors.h"
+#include "events.h"
 
 TestUI::TestUI(int num_cols, int num_rows, float aspect)
     : UI(), // explicit call base class dflt constructor
       ui_grid(0, 0, 30.f * aspect, 80.f, num_cols*num_rows, num_cols, 9, 9, color::indian_red),
       rect(0, 0, 9, 9, color::windwaker_green),
+      bottom_rect(0, 0, 9, 2, color::black),
       info_panel(0, 80.f),
-      info_rect(0, 0, 30.f * aspect, 20.f, color::ocean_blue) {
+      info_rect(0, 0, 30.f * aspect, 20.f, color::ocean_blue),
+      title_label("Select a block", 0, 0, 1.f, 1.f, color::white) {
     for (int i = 0; i < num_rows; ++i) {
         for (int j = 0; j < num_cols; ++j) {
             ui_grid.attach_at(i, j, rect);
+
+            if (i == 0 && j == 0) {
+                UIContainer* cont = new UIContainer(0, 0);
```

Comment:
These should be abstracted to a `add_construct` method to reduce code duplication. It also seems odd that you're sticking these inside the for loop when they only run for one iteration of it anyways.